fix: Add libsasl2-modules-gssapi-mit to Dockerfile

ryanhall07 opened the pull request:
<!-- Provide summary of changes -->
This shared library is needed for users of the kafka plugin + kerberos+gssapi. 

Without they get the error 

```
sasl_[ssl://xxxxxxxxnet:6668/bootstrap](ssl://xxxxxxxxnet:6668/bootstrap): Cyrus/libsasl2 is missing a GSSAPI module: make sure the libsasl2-modules-gssapi-mit or cyrus-sasl-gssapi packages are installed
```

<!-- Issue number, if available. E.g. "Fixes #31", "Addresses #42, #77" -->

Fixes https://github.com/fluent/fluent-bit/issues/10240

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [ N/A] Example configuration file for the change
- [ N/A] Debug log output from testing the change
<!--
Please refer to the Developer Guide for instructions on building Fluent Bit with Valgrind support:
https://github.com/fluent/fluent-bit/blob/master/DEVELOPER_GUIDE.md#valgrind
Invoke Fluent Bit and Valgrind as: $ valgrind --leak-check=full ./bin/fluent-bit <args>
-->
- [N/A ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [ N/A] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ N/A] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**
<!-- Docs can be edited at https://github.com/fluent/fluent-bit-docs -->
- [ N/A] Documentation required for this feature

<!--  Doc PR (not required but highly recommended) -->

**Backporting**
<!--
PRs targeting the default master branch will go into the next major release usually.
If this PR should be backported to the current or earlier releases then please submit a PR for that particular branch.
-->
- [ N/A] Backport to latest stable release.

<!--  Other release PR (not required but highly recommended for quick turnaround) -->
----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

patrick-stephens commented:
We should really update the sanity tests as well to start verifying these runtime dependencies ideally too.

patrick-stephens commented:
Does this resolve #10240? Are there any others we should add now?

patrick-stephens commented on dockerfiles/Dockerfile:
```diff
@@ -132,6 +132,7 @@ RUN echo "deb http://deb.debian.org/debian bookworm-backports main" >> /etc/apt/
     libpsl5 \
     libbrotli1 \
     libsasl2-2 \
+    libsasl2-modules-gssapi-mit \
```
I presume this requires no additional build time dependencies?

stoksc replied:
No, builds work fine without anything additional and with this flb finds the library

stoksc commented:
@patrick-stephens I will try to add a simple test.

FWIW, I have a true end to end test for this that runs Kafka and Kerberos that works, but it is complicated.

stoksc committed:
dockerfile: add libsasl2-modules-gssapi-mit and krb5-user

libsasl2-modules-gssapi-mit is needed by librdkafka to use the
following configurations for authentication with Kafka:
```
rdkafka.security.protocol    SASL_SSL
rdkafka.sasl.mechanism       GSSAPI
```

krb5-user is needed by librdkafka to invoke `kinit` to obtains
ticket-granting tickets

These changes _cannot_ be added to the production image target
because librdkafka uses the `system` call to invoke `kinit`,
the distroless base does not have a shell, and we are not
going to add it.

Signed-off-by: Ryan Hall <ryanhall07@gmail.com>
Signed-off-by: Bradley Laney <bradley.laney@chronosphere.io>

stoksc commented:
I had to unwind the changes to the `production` target because distroless doesn't have a shell and librdkafka uses the `system` call to invoke `kinit`.

patrick-stephens commented:
> A question though, why is there so much divergence between production and debug's packages?

Debug needs a lot of things to help you debug things, production should only have what is required to run. In addition, "just" adding a shell and package managers requires another entire stack of dependencies - one of the reasons for distroless.

patrick-stephens commented:
We have the integration tests if that's the only way but was hoping a simple dry run should work
